### PR TITLE
Update github actions

### PR DESCRIPTION
**Pull request by DevScyu**

### CODE REVIEW

**Review by kristofbolyai — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        uses: actions/setup-java@v2
+        with:
+          distribution: "temurin"
+          java-version: 17
+          cache: "gradle"
+
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build -x spotlessCheck --no-daemon
```
Comment (magicus) [inline .github/workflows/build.yml]: Argh! No trailing newline! :)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+name: build
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+  pull_request_target:
```
Comment (magicus) [inline .github/workflows/build.yml]: If you add `workflow_dispatch`, we can start the action manually.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

**Comment by kristofbolyai** [comment]
@DevScyu are you working on this, or should I pick this up?

**Review by magicus — DISMISSED** [review]
Sorry @DevScyu for hijacking this but I think this should go in...

**Review by kristofbolyai — DISMISSED** [review]

**Comment by kristofbolyai** [comment]
Good thing build task is broken here :P.

**Comment by kristofbolyai** [comment]
I guess we need @DevScyu to merge this.

**Comment by magicus** [comment]
Build does not seem to report back. I wonder if this is a temporary hiccup or a sign that the fix is broken...?

**Comment by kristofbolyai** [comment]
> Build does not seem to report back. I wonder if this is a temporary hiccup or a sign that the fix is broken...?

Does the PR build task use the new file? Can we test that by renaming the tasks?

**Comment by magicus** [comment]
Hm, I guess not:

```
on:
  push:
    branches:
      - main
```

I got confused since on the OpenJDK project we have configured our GHA actions to run on PRs.

**Review by kristofbolyai — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Pull request merged**